sdk/tables/azure-data-tables/azure/data/tables/_serialize.py

```diff
@@ -34,6 +34,8 @@ def _get_match_headers(etag, match_condition):
 
 def _prepare_key(keyvalue):
     """Duplicate the single quote char to escape."""
+    if isinstance(keyvalue, UUID):
+        keyvalue = str(keyvalue)
```

Comment:
I don't think we need the `isinstance` check. I think we could just update line 40, and it would work:
`return str(keyvalue).replace("'", "''")`

Reply:
After did this change, I realize it will convert any type, instead of UUID type, to string. Like in this test, it convert int to string, so that it doesn't throw `TypeError` now. https://github.com/Azure/azure-sdk-for-python/blob/beb2b51b865587987e9b7c55537255b7bda14552/sdk/tables/azure-data-tables/tests/test_table_entity.py#L2255-L2270
Should we allow all types conversion?